src/main/java/org/terasology/ligthandshadow/barrier/MagicDomeSystem.java

```diff
@@ -28,119 +18,135 @@
 import org.terasology.engine.entitySystem.systems.UpdateSubscriberSystem;
 import org.terasology.engine.logic.characters.CharacterImpulseEvent;
 import org.terasology.engine.logic.characters.CharacterMoveInputEvent;
-import org.terasology.engine.logic.console.commandSystem.annotations.Command;
 import org.terasology.engine.logic.location.LocationComponent;
 import org.terasology.engine.registry.In;
-import org.terasology.engine.utilities.random.FastRandom;
-import org.terasology.engine.world.RelevanceRegionComponent;
-import org.terasology.itemRendering.components.AnimateRotationComponent;
+import org.terasology.lightandshadowresources.components.LASTeamComponent;
+import org.terasology.ligthandshadow.componentsystem.LASUtils;
+import org.terasology.ligthandshadow.componentsystem.events.BarrierActivateEvent;
+import org.terasology.ligthandshadow.componentsystem.events.BarrierDeactivateEvent;
+import org.terasology.module.inventory.systems.InventoryManager;
 
-@RegisterSystem
+@RegisterSystem(RegisterMode.AUTHORITY)
 public class MagicDomeSystem extends BaseComponentSystem implements UpdateSubscriberSystem {
-
+    private static final int WORLD_RADIUS = 20;
     private static final Logger logger = LoggerFactory.getLogger(MagicDomeSystem.class);
-
-    private static final int WORLD_RADIUS = 500;
-
     @In
     private EntityManager entityManager;
+    @In
+    private InventoryManager inventoryManager;
 
-    private Vector3f lastPos = new Vector3f();
-    private EntityRef magicDomeEntity = EntityRef.NULL;
     private float updateDelta;
-    private FastRandom random = new FastRandom();
+    private Vector3f df = new Vector3f();
+    private Vector3f temp = new Vector3f();
+    private EntityRef redBarrier = EntityRef.NULL;
+    private EntityRef blackBarrier = EntityRef.NULL;
 
-    @Override
-    public void postBegin() {
-        //toggleDome();
-    }
 
-    @Command(shortDescription = "Activate/Deactivate dome barrier", helpText = "Activates or deactivates the dome barrier around the world", runOnServer = true)
-    public String dome() {
-        toggleDome();
-        return "Toggled dome.";
+    @ReceiveEvent
+    public void activateBarrier(BarrierActivateEvent event, EntityRef entity) {
```

Comment:
```suggestion
    public void activateBarriers(BarrierActivateEvent event, EntityRef entity) {
```

Reply:
This definitely needs some Javadoc.

- it activates the barriers for both teams
- but only if **both** barrier entities don't exist
- it will complete without any effect or log trace if any of the barriers is not the null entity